



> CE126


> **Assignment: Perform SVM with PCA operation on Breast Cancer Dataset and Iris Dataset.**









In [36]:
import numpy as np 
import matplotlib.pyplot as plt 
from sklearn.decomposition import PCA 
import pandas as pd 
import math 
import random 
from sklearn import datasets

In [37]:
dataset = datasets.load_breast_cancer()
#load data

In [38]:
dataset.data.shape


(569, 30)

In [39]:
dataset.target.shape


(569,)

In [40]:
dataset.feature_names


array(['mean radius', 'mean texture', 'mean perimeter', 'mean area',
       'mean smoothness', 'mean compactness', 'mean concavity',
       'mean concave points', 'mean symmetry', 'mean fractal dimension',
       'radius error', 'texture error', 'perimeter error', 'area error',
       'smoothness error', 'compactness error', 'concavity error',
       'concave points error', 'symmetry error',
       'fractal dimension error', 'worst radius', 'worst texture',
       'worst perimeter', 'worst area', 'worst smoothness',
       'worst compactness', 'worst concavity', 'worst concave points',
       'worst symmetry', 'worst fractal dimension'], dtype='<U23')

In [41]:
dataset.target_names

array(['malignant', 'benign'], dtype='<U9')

In [42]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(dataset.data, dataset.target, test_size = 0.2,random_state=126)


**SVM Without Applying Dimensinality Reduction.**



In [43]:

from sklearn import svm,metrics
model = svm.SVC(kernel='linear')
model.fit(x_train,y_train)
pred = model.predict(x_test)

In [44]:
print("Accuracy:",metrics.accuracy_score(y_test, pred))
print("Precision:",metrics.precision_score(y_test, pred))
print("Recall:",metrics.recall_score(y_test,pred))

Accuracy: 0.956140350877193
Precision: 0.9367088607594937
Recall: 1.0


**SVM With Dimensinality Reduction.**

In [45]:
pca = PCA(n_components=25) 
pca_fit = pca.fit(dataset.data)

In [46]:
pca_tran = pca_fit.transform(dataset.data)

In [47]:
column_list=dataset.feature_names
print(type(column_list))

column_list=list(column_list)
print(type(column_list))

dataPCA = pd.DataFrame(data = pca_tran, columns = column_list[:25])

<class 'numpy.ndarray'>
<class 'list'>


In [48]:
x_tr, x_ts, y_tr, y_ts = train_test_split(dataPCA, dataset.target, test_size = 0.2,random_state=126)

In [49]:
x_tr.shape

(455, 25)

In [50]:
y_tr.shape


(455,)

In [51]:
model1 = svm.SVC(kernel='linear')
model1.fit(x_tr,y_tr)
pred1 = model1.predict(x_ts)

In [52]:
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_ts, pred1))
print("Precision:",metrics.precision_score(y_ts, pred1))
print("Recall:",metrics.recall_score(y_ts, pred1))

Accuracy: 0.9473684210526315
Precision: 0.925
Recall: 1.0


For random state 126 i am getting same metrices parameter.


---

Accuracy: 0.956140350877193

Precision: 0.9367088607594937

Recall: 1.0

---



Accuracy: 0.9473684210526315

Precision: 0.925

Recall: 1.0

---

